usr/lib/anon-shared-helper-scripts/torrc-parser

```diff
@@ -0,0 +1,107 @@
+function get_tor_start_command() {
+    ## Tor use systemd to start
+    tor_start_command=$(grep -i "ExecStart=" /lib/systemd/system/tor@default.service | sed s/ExecStart=//g)
+
+    if [ -z "$tor_start_command" ]; then
+        echo "[error] Failed to extra commands used to start Tor." >&2
+        return 1
+    fi
+
+    ## Tor is run as root, we need to simulate this, too.
+    tor_start_command="sudo $tor_start_command"
+}
+
+## extra_specified_torrc will extract torrc files specified in
+## $tor_start_command. Specifically, it will set "$default_torrc" and
+## "$f_torrc"
+function extra_specified_torrc() {
+    ## extract default_torrc file and -f torrc file
+    if [[ "$tor_start_command" =~ (--defaults-torrc )([^ ]*) ]]; then
+	default_torrc="${BASH_REMATCH[2]}"
+    fi
+
+    if [[ "$tor_start_command" =~ (-f )([^ ]*) ]]; then
+	f_torrc="${BASH_REMATCH[2]}"
+    fi
+}
+
+## one common call to parser is:
+## parser "$default_torrc" "$f_torrc"
+
+## parser() will write the parsing result to the variable
+## $torrc_all_in_one
+function parser() {
+    for abs_path in "$@"; do
+	local abs="$abs_path"
+        torrc_all_in_one+="===> Parsing $abs_path"$'\n'
+	if [[ -f  "$abs_path" ]]; then
+	    torrc_files+=("$abs_path")
+	    local file_content
+            file_content="$(cat "$abs")"
+	    IFS=$'\n'
+            # only show comments in torrc files when $VERY_VERBOSE is "1"
+	    for line in $file_content; do
+                if [[ "$line" =~ ^[[:space:]]*# && "$VERY_VERBOSE" != "1" ]]; then
+                    continue
+                else
+                    torrc_all_in_one+="$line"$'\n'
+                fi
+
+		if [[ "$line" =~ ^[[:space:]]*%include ]]; then
+                    parser "$(echo "$line" | awk '{print $2;}')"
+		fi
+	    done
+	elif [[ -d "$abs_path" ]]; then
+		 parser "$(find "$abs_path" -mindepth 1 | sort -g)"
+        fi
+        torrc_all_in_one+="===> Done parsing $abs"$'\n'
+    done
+}
+
+function parsing_order_verbose(){
+    echo "The Tor configuration files are parsed in such order: "
+    echo -n "$torrc_all_in_one"
+}
+
+function used_torrc_files(){
+    echo "${#torrc_files[@]} files are used as Tor configuration files: "
+    echo "${torrc_files[@]}"
+}
+
+function unknown_option_specifier(){
+    for unknown_option in "$@"; do
+	echo "The following files contain unkown option '$unknown_option': "
+	grep -iHnE --color "^([[:space:]])*$unknown_option" "${torrc_files[@]}"
+	echo "Please modify these files to fix the problem."
+    done
+}
+
+## this call sets the final value of DisableNetwork to $TOR_ENABLED
+## $TOR_ENABLED is used by tor_enabled_check in the whonixcheck
+## Note that it doesn't mean it is the value Tor is using because the
+## value can always be changed via Tor controll port.
+function value_of_disable_network(){
+    TOR_ENABLED=$(awk 'BEGIN{IGNORECASE = 1}{if ($1 == "DisableNetwork") value=$2}END{print value}')  <<< echo "$torrc_all_in_one" | sed 's/#.*//g'
```

Comment:
That is quite difficult to grasp. It also requires basic understanding of `awk`. Could you simplify that please? Perhaps something like it was before? Pseudo code:

```
for line in $torrc_all_in_one ; do
    line=to lower case line
    if [ "$line" = "disablenetwork 0" ]; then
      ...
    elif [ "$line" = "disablenetwork 1" ]; then
      ...
    fi
done
```

Last line would prevail in the script (which is also what would win in torrc.d parsed by Tor).